app/containers/markdown/index.tsx

```diff
@@ -86,271 +84,244 @@ const emojiCount = (str: string) => {
+					code: renderCodeInline,
+					link: renderLink,
+					image: renderImage,
+					atMention: renderAtMention,
+					emoji: renderEmoji,
+					hashtag: renderHashtag,
+
+					paragraph: renderParagraph,
+					heading: renderHeading,
+					codeBlock: renderCodeBlock,
+					blockQuote: renderBlockQuote,
+
+					list: renderList,
+					item: renderListItem,
+
+					hardBreak: renderBreak,
+					thematicBreak: renderBreak,
+					softBreak: renderBreak,
+
+					htmlBlock: renderText,
+					htmlInline: renderText,
+
+					table: renderTable,
+					table_row: renderTableRow,
+					table_cell: renderTableCell
+				},
+				renderParagraphsInLists: true
+			}),
+		[]
+	);
+
+	useEffect(() => {
+		if (!isNewMarkdown() && msg) {
+			renderer.current = createRenderer();
+			forceUpdate();
 		}
-	}
+	}, [createRenderer, isNewMarkdown, msg]);
 
-	createRenderer = () =>
-		new Renderer({
-			renderers: {
-				text: this.renderText,
-
-				emph: Renderer.forwardChildren,
-				strong: Renderer.forwardChildren,
-				del: Renderer.forwardChildren,
-				code: this.renderCodeInline,
-				link: this.renderLink,
-				image: this.renderImage,
-				atMention: this.renderAtMention,
-				emoji: this.renderEmoji,
-				hashtag: this.renderHashtag,
-
-				paragraph: this.renderParagraph,
-				heading: this.renderHeading,
-				codeBlock: this.renderCodeBlock,
-				blockQuote: this.renderBlockQuote,
-
-				list: this.renderList,
-				item: this.renderListItem,
-
-				hardBreak: this.renderBreak,
-				thematicBreak: this.renderBreak,
-				softBreak: this.renderBreak,
-
-				htmlBlock: this.renderText,
-				htmlInline: this.renderText,
-
-				table: this.renderTable,
-				table_row: this.renderTableRow,
-				table_cell: this.renderTableCell
-			},
-			renderParagraphsInLists: true
-		});
-
-	get isNewMarkdown(): boolean {
-		const { md, enableMessageParser } = this.props;
-		return !!enableMessageParser && !!md;
+	if (!msg) {
+		return null;
 	}
 
-	renderText = ({ context, literal }: { context: []; literal: string }) => {
-		const { numberOfLines, style = [] } = this.props;
-		const defaultStyle = [this.isMessageContainsOnlyEmoji ? styles.textBig : {}, ...context.map(type => styles[type])];
-		return (
-			<Text accessibilityLabel={literal} style={[styles.text, defaultStyle, ...style]} numberOfLines={numberOfLines}>
-				{literal}
-			</Text>
-		);
-	};
-
-	renderCodeInline = ({ literal }: TLiteral) => {
-		const { theme, style = [] } = this.props;
-		return (
-			<Text
-				style={[
-					{
-						...styles.codeInline,
-						color: themes[theme].bodyText,
-						backgroundColor: themes[theme].bannerBackground,
-						borderColor: themes[theme].bannerBackground
-					},
-					...style
-				]}>
-				{literal}
-			</Text>
-		);
-	};
+	const formattedMessage = formatHyperlink(formatText(msg));
+	const ast = mergeTextNodes(parser.parse(formattedMessage));
+	isMessageContainsOnlyEmoji.current = isOnlyEmoji(formattedMessage) && emojiCount(formattedMessage) <= 3;
```

Comment:
I think you can move these lines 163 -165 to before the line 325. Because if isNewMarkdown is true, you don't need to waste time parsing the message with the lib. WDYT?